packages/nuxt/src/app/plugins/router.ts

```diff
@@ -222,8 +222,8 @@ export default defineNuxtPlugin<{ route: Route, router: Router }>((nuxtApp) => {
   nuxtApp.hooks.hookOnce('app:created', async () => {
     router.beforeEach(async (to, from) => {
       to.meta = reactive(to.meta || {})
-      if (nuxtApp.isHydrating) {
-        to.meta.layout = initialLayout.value ?? to.meta.layout
+      if (nuxtApp.isHydrating && initialLayout.value !== undefined && initialLayout.value !== null) {
```

Comment:
nice find.

would we be better off checking if `to.meta.layout` is writeable with the `isReadonly` helper from vue?

```suggestion
      if (nuxtApp.isHydrating && !isReadonly(to.meta.layout) {
```